src/library/export/exportrequest.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <QSet>
+
+#include "library/trackset/crate/crateid.h"
+
+namespace mixxx {
+
+/// The ExportRequest struct is the abstract base class for all types of request
+/// to export the Mixxx library to some external location.  An export to a
+/// specific kind of external location is likely to need additional information
+/// to be specified in the request, which can be achieved by sub-classing this
+/// struct.
+struct ExportRequest {
+    virtual ~ExportRequest() = default;
+
+    /// Indicates whether to export only selected crates (if set to true), or
+    /// whether to export the whole Mixxx library (if set to false).
+    bool exportSelectedCrates;
```

Comment:
Please initialize `exportSelectedCrates`. Use direct initialization, no constructor needed.

Reply:
Doesn't preventive initialization (with a dummy) hide compiler warnings about code branches with unitialized variable usage?

Reply:
It is good practice to not leave objects partially uninitialized after (default) construction. I doubt that the compiler will detect this kind of uninitialized usage if you forget to assign some of them.

Reply:
Btw, `exportSelectedCrates` would be redundant if you interpret *no selected crates = complete library*

Reply:
The boolean has now been removed completely, and instead an empty `crateIdsToExport` set implies that the whole library is to be exported.